tests/bgp/test_startup_tsa_tsb_service.py

```diff
@@ -304,8 +320,8 @@ def test_tsa_tsb_service_with_dut_abnormal_reboot(duthosts, localhost, enum_rand
     dut_nbrhosts = nbrhosts_to_dut(duthost, nbrhosts)
     dut_ip = duthost.mgmt_ip
     # Ensure that the DUT is not in maintenance already before start of the test
-    pytest_assert(TS_NORMAL == get_traffic_shift_state(duthost),
-                  "DUT is not in normal state")
+    if TS_NORMAL != get_traffic_shift_state(duthost):
+        pytest.skip("Test skipped since DUT is not in normal state")
```

Comment:
Hi @arlakshm, We have seen that the testcase fails due to this reason, the actual testcase has not even executed but shows up as failure, which according to me is a wrong behavior. We can xfail if you feel we should not skip the test.

Reply:
@vperumal, I suspect the previous test is not working as expected and leaving the testbed in bad state. I think we need to fix that iso modifying this test

Reply:
These kind of issues at the start and end of the _startup_tsa_tsb_ tests have been resolved and taken care already as part of the PR #13290, which is still in review stage and yet to be merged unfortunately.

Reply:
Implementing sanjair's solution as an autouse fixture where it asserts instead of skip would be ideal for consistency between tests and accurate test reporting

Reply:
The assert wrongly states the current testcase as failing, wherein the actual test has not even been executed. Hence I changed it to skip. We can change it to xfail if that everyone feels we need to fail it. Because failing a testcase even before it started seems wrong to me

Reply:
Agreed failing is not the most accurate state - my main concern with skipping is that they should be reserved for testcases that don't need a result (testcases not valid for T2 topo etc.). Skipping the test could cause it to be overlooked. Not sure about everyone else's opinion?

Reply:
Agreed. we can make it xfail if everyone is comfortable with that approach

Reply:
We have sanity_check, why would the testbed be in bad state?

Reply:
@yejianquan this is specifically for TSA/TSB state, does sanity-check cover this?